src/compiler/checker.ts

```diff
@@ -39048,7 +39048,7 @@ namespace ts {
                 const properties = getPropertiesOfType(getTypeWithThisArgument(base, type.thisType));
                 for (const prop of properties) {
                     const existing = seen.get(prop.escapedName);
-                    if (existing && !isPropertyIdenticalTo(existing, prop)) {
+                    if (existing && prop.parent === existing.parent && !isPropertyIdenticalTo(existing, prop)) {

```

Comment:
```suggestion
                    if (existing && prop.parent === existing.parent) {

```

You can do away with the identity check entirely - the `.parent` similarity check is sufficient. Dunno why I didn't think of that originally.

Reply:
@weswigham Thanks.